### PR TITLE
[SSR] Insert dehydrated data directly into HTML doc after main markup.

### PR DESCRIPTION
**Pull request by alloy**
This reverts commit 42cbf810a0c7f294042110a91a54d0af5668fba6, which reverted https://github.com/artsy/force/pull/3108.

Depends on https://github.com/artsy/reaction/pull/1582 to be released first.

### CODE REVIEW

File: src/desktop/apps/collect2/server.tsx [hunk 1]
```diff
@@ -6,14 +6,13 @@ import React from "react"
 import { Meta } from "./meta"
 import { buildServerAppContext } from "desktop/lib/buildServerAppContext"
 
-const app = (module.exports = express())
+export const app = (module.exports = express())
```
Comment (damassi) [inline src/desktop/apps/collect2/server.tsx]: I think `module.exports` needs to be removed here

File: src/desktop/apps/collections/server.tsx [hunk 2]
```diff
@@ -5,13 +5,13 @@ import { routes } from "reaction/Apps/Collections/routes"
 import express from "express"
 import { buildServerAppContext } from "desktop/lib/buildServerAppContext"
 
-const app = (module.exports = express())
+export const app = (module.exports = express())
```
Comment (damassi) [inline src/desktop/apps/collections/server.tsx]: Same here

**Comment by damassi** [comment]
Couple minor comments but merge when ready (also thank you for renaming files to `.tsx`)

File: src/desktop/apps/collect2/server.tsx (hunk 1 above)
Comment (alloy) [reply src/desktop/apps/collect2/server.tsx]: Ugh, yeah, good catch

**Comment by alloy** [comment]
All looking good locally, gonna move to staging.

**Pull request merged**